apps/blockchain/lib/blockchain/backend.ex

```diff
@@ -102,79 +100,50 @@ defmodule Blockchain.Backend do
     {module, func, args}
   end
 
-  @spec find_backend(backend(), state()) :: resp({:ok, pid()} | {:not_started, mfargs()})
-  defp find_backend({backend, _id} = backend_spec, {_, handlers, registry}) do
-    case handlers do
-      %{^backend => mfargs} ->
-        case registry do
-          %{^backend_spec => pid} ->
-            {:ok, pid}
-
-          _ ->
-            {:not_started, mfargs}
-        end
-
-      _ ->
-        {:error, :no_handler}
-    end
-  end
-
   @spec ensure_backend_started(atom() | backend(), state()) :: resp({:ok, server(), state()})
   defp ensure_backend_started(backend, state) when is_atom(backend) do
     ensure_backend_started({backend, nil}, state)
   end
 
-  defp ensure_backend_started({backend, _id} = backend_spec, {supervisor, h, registry} = state) do
-    case find_backend(backend_spec, state) do
-      {:ok, pid} ->
-        {:ok, pid, state}
-
-      {:not_started, mfargs} ->
+  defp ensure_backend_started({backend, _id} = backend_spec, {supervisor, handlers}) do
+    case handlers do
+      %{^backend => mfargs} ->
         retval =
           DynamicSupervisor.start_child(supervisor, %{
             id: backend_name(backend_spec),
-            start: mfargs
+            start: mfargs,
+            restart: :temporary
           })
 
         case retval do
           {:ok, pid} ->
-            {:ok, pid, {supervisor, h, Map.put(registry, backend_spec, pid)}}
+            {:ok, pid}
 
           error ->
             :ok = Logger.error("Failed to start backend for #{backend}: #{inspect(error)}")
             {:error, :start_failed}
         end
 
-      {:error, error_code} ->
-        {:error, error_code}
+      _ ->
+        {:error, :no_handler}
     end
   end
 
   ## Callbacks
   ##
 
-  @doc """
-  Handles the start_backend call from the client API start_backend/2.
-  """
-  @spec handle_call({:start_backend, backend()}, from(), state()) :: reply(:ok)
-  def handle_call({:start_backend, backend_spec}, _from, state) do
-    case ensure_backend_started(backend_spec, state) do
-      {:ok, _pid, state1} ->
-        {:reply, :ok, state1}
-
-      error ->
-        {:reply, error, state}
-    end
-  end
-
   @doc """
   Handles the call call from the client API call/4.
   """
   @spec handle_call({:call, backend(), call()}, from(), state()) :: reply({:ok, any()})
   def handle_call({:call, backend_spec, func_spec}, _from, state) do
     case ensure_backend_started(backend_spec, state) do
-      {:ok, pid, state1} ->
-        {:reply, {:ok, GenServer.call(pid, func_spec)}, state1}
+      {:ok, pid} ->
+        try do
```

Comment:
Not too familiar with this code. Would it suppress exceptions from `GenServer.call(pid, func_spec)` without logging?

Reply:
It shouldn’t, because we didn’t catch anything. `after` just ensure the code always get executed regardless whether the code in `try` block raises or not.